pkg/operator/controllers/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -37,45 +37,60 @@ const (
 	enabled        string = "aro.machinehealthcheck.enabled"
 )
 
-type Reconciler struct {
-	log *logrus.Entry
+type MachineHealthCheckReconciler struct {
+	base.AROController
 
 	dh dynamichelper.Interface
-
-	client client.Client
 }
 
-func NewReconciler(log *logrus.Entry, client client.Client, dh dynamichelper.Interface) *Reconciler {
-	return &Reconciler{
-		log:    log,
-		dh:     dh,
-		client: client,
+func NewMachineHealthCheckReconciler(log *logrus.Entry, client client.Client, dh dynamichelper.Interface) *MachineHealthCheckReconciler {
+	return &MachineHealthCheckReconciler{
+		AROController: base.AROController{
+			Log:    log,
+			Client: client,
+			Name:   ControllerName,
+		},
+		dh: dh,
 	}
 }
 
-func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
-	instance := &arov1alpha1.Cluster{}
-	err := r.client.Get(ctx, types.NamespacedName{Name: arov1alpha1.SingletonClusterName}, instance)
+// Reconcile watches MachineHealthCheck objects, and if any changes,
+// reconciles the associated ARO MachineHealthCheck object
+func (r *MachineHealthCheckReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	instance, err := r.GetCluster(ctx)
+
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
 	if !instance.Spec.OperatorFlags.GetSimpleBoolean(enabled) {
-		r.log.Debug("controller is disabled")
+		r.Log.Debug("controller is disabled")
 		return reconcile.Result{}, nil
 	}
 
-	r.log.Debug("running")
+	r.Log.Debug("running")
 	if !instance.Spec.OperatorFlags.GetSimpleBoolean(managed) {
+		r.SetProgressing(ctx, "MHC and it's alerts are disabled because this cluster is not MHC managed.")
+
 		err := r.dh.EnsureDeleted(ctx, "MachineHealthCheck", "openshift-machine-api", "aro-machinehealthcheck")
 		if err != nil {
+			r.Log.Error(err)
+			r.SetDegraded(ctx, err)
+			r.ClearProgressing(ctx)
```

Comment:
Ordering?
```suggestion
			r.ClearProgressing(ctx)
			r.SetDegraded(ctx, err)
```

Reply:
Order shouldn't actually matter but I agree with the change for consistency's sake

Reply:
The way it is written here if the MHC alerts and Prometheus Rule aren't deleted then set Degraded so consider this way if they are deleted so it will not be in degraded but it has to clear the Progressing field mentioned above where it says not MHC Managed for both MHC alerts and Prometheus Rules and this clear Progressing should be for both of them individually.